### PR TITLE
Checkboxes - new IA

### PR DESCRIPTION
**Pull request by chri2547**
Image below updated on 7/25, pm. Ready for content-lgtm.

image

### CODE REVIEW

File: _component_design/checkboxes.md [hunk 1]
```diff
@@ -18,25 +18,92 @@ helix-ui-javascript: true
...
+<div class="hxRow"  markdown="1">
+
+{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-8-md hxSpan-8-lg" %}
+
+## When to use
+
+Use checkboxes in the following situations:
+
+- When the user needs to answer a question or wants to change a control panel setting. Do not use a checkbox to trigger another process or workflow.
+- When the user can select any number of choices, including zero, one, or several choices. If a user can only select one option, use a radio button or a drop-down list.
```
Comment (palindromebart) [inline _component_design/checkboxes.md]: s/dropdown list/input selector

File: _component_design/checkboxes.md [hunk 2]
```diff
@@ -18,25 +18,92 @@ helix-ui-javascript: true
...
+- When the user can turn on or turn off a single option. For example, use a single checkbox when a user accepts the terms of an End User License Agreement (EULA).
+- When it is important for the user to compare all options simultaneously. If it is not important for the user to compare all options at the same time, consider using a multi-select drop-down menu. A multi-select drop-down menu uses less space than a checkbox group (but introduces greater cognitive load) and is more difficult to navigate for users who have difficulty making precise mouse movements.
```
Comment (palindromebart) [inline _component_design/checkboxes.md]: s/multi-select drop-down menu/input selector

File: _component_design/checkboxes.md [hunk 3]
```diff
@@ -18,25 +18,92 @@ helix-ui-javascript: true
...
+
+<div class="hxRow"  markdown="1">
+
+{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-8-md hxSpan-8-lg" %}
+
+## Best practices
+
+Consider the following best practices when you construct a checkbox group or single checkbox:
+
+- List checkbox options vertically, with one choice per line. If you must use a horizontal layout, consider using a checkbox button group instead.
```
Comment (palindromebart) [inline _component_design/checkboxes.md]: do you mean "consider using a selector strip instead"?

File: _component_design/checkboxes.md [hunk 4]
```diff
@@ -18,25 +18,92 @@ helix-ui-javascript: true
...
+- Construct a checkbox input so that it is independent from all other inputs in the list. A selected checkbox does not de-select any other checkboxes.
+- When a user needs to manage a group of items, add checkboxes to a list to allow for bulk actions.
+- In most cases, there is no default checkbox selected. However, if Rackspace recommends an option, make that option the default selection.
+- Ensure that you separate groups and checkbox options from other groups and options on the same page.
+- With sub-heads, users might misunderstand each sub-group as a separate set of options. With checkboxes, each box is an independent choice.
```
Comment (palindromebart) [inline _component_design/checkboxes.md]: i'm confused about this instruction. what's a sub-head?

File: _component_design/checkboxes.md [hunk 5]
```diff
@@ -18,25 +18,92 @@ helix-ui-javascript: true
...
+
+## Best practices
+
+Consider the following best practices when you construct a checkbox group or single checkbox:
+
+- List checkbox options vertically, with one choice per line. If you must use a horizontal layout, consider using a checkbox button group instead.
+- Construct a checkbox input so that it is independent from all other inputs in the list. A selected checkbox does not de-select any other checkboxes.
+- When a user needs to manage a group of items, add checkboxes to a list to allow for bulk actions.
+- In most cases, there is no default checkbox selected. However, if Rackspace recommends an option, make that option the default selection.
+- Ensure that you separate groups and checkbox options from other groups and options on the same page.
```
Comment (palindromebart) [inline _component_design/checkboxes.md]: what does this mean?

File: _component_design/checkboxes.md [hunk 6]
```diff
@@ -18,25 +18,92 @@ helix-ui-javascript: true
...
+- With sub-heads, users might misunderstand each sub-group as a separate set of options. With checkboxes, each box is an independent choice.
+
+Consider the following best practices when you label a checkbox option:
+
+- Use positive and active wording for option labels. Avoid negations such as **Don't send me more email**. Negative option labels indicate that the user must select the checkbox in order for something *not to happen*.
+- Write an option label so that a user understands what happens when the checkbox is selected, and what will *not* happen if the checkbox is not selected. If you can't write a clear label, use two radio buttons instead: one radio button for turning the feature on and one radio button for turning the feature off. Ensure you write clear labels for each of the two cases.
```
Comment (palindromebart) [inline _component_design/checkboxes.md]: "consider using..."

File: _component_design/checkboxes.md [hunk 7]
```diff
@@ -218,17 +306,16 @@ In the case of an error, always include an error message below the checkbox grou
 
 {% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-4-md hxSpan-4-lg" %}
 
-#### Indeterminate
+### Indeterminate
```
Comment (palindromebart) [inline _component_design/checkboxes.md]: we should prob treat this section more like we are for the switch in a table. Indeterminate checkboxes should only be used in a table and only for bulk selection.

File: _component_design/checkboxes.md [hunk 8]
```diff
@@ -218,17 +306,16 @@ In the case of an error, always include an error message below the checkbox grou
...
 
-The indeterminate state is only used with bulk selections to indicate that one or more, but not all options, have been selected. Refer to the [tables]({{site.baseurl}}/components/tables.html) page for more information.
+Use an indeterminate state with nested checkboxes that provide for bulk selections. An indeterminate checkbox indicates that the user has selected one or more, but not all, options in a nested group. See  [tables]({{site.baseurl}}/components/tables.html) for more information about checkboxes in tables.
```
Comment (palindromebart) [inline _component_design/checkboxes.md]: what's a nested checkbox? nested group?

**Comment by cyrichardson** [comment]
@chri2547 Just a few suggestions on the text: 
[Helix-check-boxes_cyrEdits.docx](https://github.com/rackerlabs/design-system/files/2215299/Helix-check-boxes_cyrEdits.docx)

File: _component_design/checkboxes.md [hunk 9]
```diff
@@ -218,17 +304,16 @@ In the case of an error, always include an error message below the checkbox grou
 
 {% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-4-md hxSpan-4-lg" %}
 
-#### Indeterminate
+## Indeterminate checkboxes
 
-The indeterminate state is only used with bulk selections to indicate that one or more, but not all options, have been selected. Refer to the [tables]({{site.baseurl}}/components/tables.html) page for more information.
+Use an indeterminate state with a nested group that provides for bulk selection. An indeterminate checkbox indicates that the user has selected one or more, but not all, options in a nested group. See  [tables]({{site.baseurl}}/components/tables.html) for more information about checkboxes in tables.
```
Comment (palindromebart) [inline _component_design/checkboxes.md]: OOOK, this still needs work. try this:

Use an indeterminate state only in table header checkbox. An indeterminate checkbox indicates that the user has selected one or more, but not all, of the child checkboxes in the table. See  [tables]({{site.baseurl}}/components/tables.html) for more information about checkboxes in tables.

File: _component_design/checkboxes.md [hunk 10]
```diff
@@ -18,25 +18,90 @@ helix-ui-javascript: true
...
+
+## When to use
+
+Use checkboxes in the following situations:
+
+- When the user needs to answer a question or wants to change a control panel setting. Do not use a checkbox to trigger another process or workflow.
+- When the user can select any number of choices, including zero, one, or several choices. If a user can only select one option, use a radio button or an input selector.
+- When the user can turn on or turn off a single option. For example, use a single checkbox when a user accepts the terms of an End User License Agreement (EULA).
+- When it is important for the user to compare all options simultaneously. If it is not important for the user to compare all options at the same time, consider using an input selector. An input selector uses less space than a checkbox group (but introduces greater cognitive load), and is more difficult to navigate for users who have difficulty making precise mouse movements.
+
```
Comment (palindromebart) [inline _component_design/checkboxes.md]: add a when to use about when a user needs to select a row in a table to act of that item.

File: _component_design/checkboxes.md [hunk 11]
```diff
@@ -79,29 +161,38 @@ When a user needs to manage a group of items, checkboxes may be added to a list
...
 
-#### Labeling checkboxes
+### Enabled
 
-- Use positive and active wording for checkbox labels. Avoid negations such as "Don't send me more email". Negative labels on checkboxes would mean that the user would have to check the box in order for something not to happen.
-- Write checkbox labels so that users know both what will happen if they check a particular box and what will happen if they leave it unchecked. If you are unable to do this, we recommend using two radio buttons: one for having the feature on and one for having it off. Ensure you write clear labels for each of the two cases.
+A user can select or deselect a checkbox in the enabled state.
 
-{% include note.html content="Allow users to select an option by clicking on either the button, the box itself, or the label; a bigger target is faster to click." %}
+See [focus states]({{site.baseURL}}/style/focus-states.html) for guidance in constructing focused checkboxes in an enabled state.
```
Comment (nimbinatus) [inline _component_design/checkboxes.md]: s/`{{site.baseURL}}`/`{{site.baseurl}}}`

File: _component_design/checkboxes.md [hunk 12]
```diff
@@ -142,20 +231,18 @@ A checkbox group consists of a section label followed by pairs of checkboxes and
...
-### States
+### Error
 
-The checkbox component may be in the following states:
+An error state can apply to checkboxes when the user selects, deselects, and hovers over a checkbox.
 
-- Selected
-- Deselected
-- Indeterminate
+See [focus states]({{site.baseURL}}/style/focus-states.html) for guidance in constructing focused checkboxes in an error state.
```
Comment (nimbinatus) [inline _component_design/checkboxes.md]: s/`{{site.baseURL}}`/`{{site.baseurl}}`

**Pull request merged**